lib/spack/spack/test/spec_semantics.py

```diff
@@ -1365,6 +1365,13 @@ def test_splice_swap_names_mismatch_virtuals(self, default_mock_concretization,
         with pytest.raises(spack.spec.SpliceError, match="virtual"):
             vt.splice(vh, transitive)
 
+    def test_adaptor_optflags(self):
+        spec = spack.concretize.concretize_one(Spec("pkg-a"))
+        assert "-Otestopt" in spec.package.compiler.opt_flags
+        # This is not set, make sure we get an empty list
+        for x in spec.package.compiler.debug_flags:
+            pass
```

Comment:
Minor tweaks to try improving clarity of the test:
```suggestion
        compiler_adaptor = spec.package.compiler
        assert compiler_adaptor.opt_flags == ["-Otestopt"]
        assert compiler_adaptor.debug_flags == []
```

Reply:
Another minor improvement could be to check against the mock `GCC.optflags` etc.

Reply:
I didn't want to test the type of the return value in this test: i.e. I intentionally did not do what you are suggesting. I think this test should be satisfied if the returned value is a sequence.